File: 프로그래머스/lv2/42842. 카펫/카펫.py
import math
def solution(brown, yellow):
    divide = []
    sum = brown + yellow
    
    for i in range(math.ceil(math.sqrt(sum)), sum + 1):
        if sum % i == 0:
            divide.append(i)
    
    for i in divide:
        x = i
        y = sum // i
        
        if (x-2) * (y-2) != yellow:
            continue
        else: 
            return [x, y]

File: 프로그래머스/2/42842. 카펫/카펫.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(int brown, int yellow) {
    vector<int> answer;
    int total = brown + yellow;

    int min_diff = 1e9;
    
    for (int i = 1; i < total / 2 + 1; i++) {
        if (total % i == 0) {
            int a = i;
            int b = total / i;

            if (a >= b && (a - 2) * (b - 2) == yellow) {
                if (min_diff > a - b) {
                    min_diff = a - b;
                    answer = {a, b};
                }
            }
        }
    }

    return answer;
}